packages/web3-providers-http/src/index.js

```diff
@@ -97,7 +97,7 @@ HttpProvider.prototype.send = function (payload, callback) {
     var request = this._prepareRequest();
 
     request.onreadystatechange = function() {
-        if (request.readyState === 4 && request.timeout !== 1) {
+        if (request.readyState === 4 && request.timeout !== 1 && (request.status >= 200 && request.status < 300)) {
```

Comment:
Can you help me understand the reasoning behind this?

Reply:
@spacesailor24 when there is a connection issue the status would be 0 which is clearly not what we want here.

(Sorry for not answering until now, somehow I didn't get any notifications)